src/common/s3.connection.cpp

```diff
@@ -0,0 +1,569 @@
+#include "s3.connection.hh"
+#include "utilities.hh"
+
+#include <miniocpp/client.h>
+
+#include <list>
+#include <sstream>
+#include <string_view>
+
+namespace zarr = acquire::sink::zarr;
+namespace common = zarr::common;
+
+common::S3Connection::S3Connection(const std::string& endpoint,
+                                   const std::string& access_key_id,
+                                   const std::string& secret_access_key)
+{
+    minio::s3::BaseUrl url(endpoint);
+    url.https = endpoint.starts_with("https");
```

Comment:
- One of our use cases is an on-prem storage cluster which supports an S3 endpoint, which does not use HTTPS. I think it's reasonable to leave HTTP support in.
- I found this [class](https://github.com/minio/minio-cpp/blob/cd4ef14955201d5ce81e54155ef0980f7c644864/include/miniocpp/providers.h#L239) in the source, but not the docs (which I think must be out of date):

```c++
struct CertificateIdentityProvider : public Provider {
 private:
  http::Url sts_endpoint_;
  std::string key_file_;
  std::string cert_file_;
  std::string ssl_cert_file_;

 public:
  CertificateIdentityProvider(http::Url sts_endpoint, std::string key_file,
                              std::string cert_file,
                              std::string ssl_cert_file = {},
                              unsigned int duration_seconds = 0);

  virtual ~CertificateIdentityProvider();

  virtual Credentials Fetch() override;
};  // struct CertificateIdentityProvider
```

Reply:
- That makes sense!
- Yeah, it’s probably outdated. It uses curl under the hood, which comes with trusted root certificates from the Mozilla store. So, I guess we don’t need to specify this manually, which would have been a pain since each OS has its own way to store this information.